docs/cookbook/entities/custom-model-accessible-per-admin-channel.rst

```diff
@@ -0,0 +1,382 @@
+.. rst-class:: plus-doc
+
+How to add a custom model accessible per admin channel?
+=======================================================
+
+In some cases you may be needing to add new model to your application that is accessible on admin site per admin channel
+
+1. Define custom model, our example will be basic on **Supplier entity**
+------------------------------------------------------------------------
+
+For making entity use guide: :doc:`here </cookbook/entities/custom-model>`.
+
+Next create channel field with relation to ``channel`` entity:
+
+.. code-block:: php
+
+    /**
+      * @var ChannelInterface
+      * @ORM\ManyToOne(targetEntity="Sylius\Plus\Entity\ChannelInterface")
+      * @ORM\JoinColumn(name="channel_id", referencedColumnName="id", nullable=true)
+      */
+     protected $channel;
+
+Create properly migration for this changes
+
+In order to be owned access to choosing only proper channel, create ChannelPerAdminChannelType:
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Form;
+
+    use Sylius\Component\Resource\Repository\RepositoryInterface;
+    use Sylius\Plus\ChannelAdmin\Application\Provider\AdminChannelProviderInterface;
+    use Sylius\Plus\Entity\ChannelInterface;
+    use Symfony\Bridge\Doctrine\Form\DataTransformer\CollectionToArrayTransformer;
+    use Symfony\Component\Form\AbstractType;
+    use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+    use Symfony\Component\Form\FormBuilderInterface;
+    use Symfony\Component\OptionsResolver\Options;
+    use Symfony\Component\OptionsResolver\OptionsResolver;
+
+    final class ChannelPerAdminChannelType extends AbstractType
```

Comment:
Can we have this form type predefined in Plus and use it somewhere?

Reply:
Yep. I think we should replace all places with `ChannelChoiceType` for `ChannelPerAdminChannelChoiceType`. This is not scope this task, but good improvement in future. Next good improvement is to add resources name as parameters in services, without decorating `ResourceChannelEnabilibityChecker` service:
Now:
```
final class ResourceChannelEnabilibityChecker implements ResourceChannelEnabilibityCheckerInterface
{
    public function forResourceName(string $resourceName): bool
    {
        return in_array($resourceName, [
            'credit_memo',
            'customer',
            'invoice',
            'order',
            'payment',
            'product',
            'product_variant',
            'return_request',
            'shipment',
        ]);
    }
}
```

It could be more generic, for example:
```
final class ResourceChannelEnabilibityChecker implements ResourceChannelEnabilibityCheckerInterface
{
    private $resources;

    public function __construct($resources)
    {
        $this->resources = array_merge(
            $resources,
            [
                'credit_memo',
                'customer',
                'invoice',
                'order',
                'payment',
                'product',
                'product_variant',
                'return_request',
                'shipment',
            ]
        );
    }


    public function forResourceName(string $resourceName): bool
    {
        return in_array($resourceName, $this->resources);
    }
```
After this changes, final application needs overwrite only service and add new entity names